naxsi_src/naxsi_utils.c

```diff
@@ -653,25 +653,24 @@ ngx_http_wlr_finalize_hashtables(ngx_conf_t *cf, ngx_http_dummy_loc_conf_t  *dlc
     }
   }
 
-  FILE *file;
-  char fname[1024];
+  FILE *file = NULL;
+  char *fname = NULL;
   if(dlc->whitelist_file)
   {
-    size_t fname_size = dlc->whitelist_file->len;
-    if (fname_size >= sizeof(fname)) {
-      fname_size = sizeof(fname);
+    fname = calloc(1024, sizeof(char));
+    if(fname !== NULL)
+    {
+      strncpy(fname, (char *)dlc->whitelist_file->data, strlen((char *)dlc->whitelist_file->data));
```

Comment:
you removed the check on strncpy like we had it before..

Reply:
Don't know how and why but this:

```
  char fname[1024];
  if(dlc->whitelist_file)
  {
    size_t fname_size = dlc->whitelist_file->len;
    if (fname_size >= sizeof(fname)) {
      fname_size = sizeof(fname);
    }
    memcpy(fname, dlc->whitelist_file->data, fname_size);
```


Worked once in a while, sometimes, yes, sometimes no.

Reply:
Now it is hardcoded at 1024? What do you suggest? @wargio is it OK?

Reply:
The len was good, but the data was corrupted, unintilaized at the end of the string @wargio

Reply:
If you want to see it, you can try on your own ... maybe you will find the root cause @wargio

Reply:
just change that to `size_t fname_size = strlen((char *)dlc->whitelist_file->data);` and `if (fname_size >= 1024) {`

Reply:
Changed